pkg/apis/app/v1alpha1/application_types.go

```diff
@@ -58,6 +59,7 @@ type ApplicationSpec struct {
 	Links []Link `json:"links,omitempty"`
 
 	// Notes contain a human readable snippets intended as a quick start for the users of the Application.
+	// Should contain plain text or plain-text-friendly markup (eg. markdown).
```

Comment:
Note can be longer form. The things that could consume this information include CLIs, desktop apps, web apps, and more. By hinting at markup like markdown without saying it we could end up with the case of markdown, asciidoc, and others being used. Then tools from varying places need to know how to handle this for interoperability.

How do you see that playing out?

Reply:
@mattfarina see my comment https://github.com/kubernetes-sigs/application/pull/48#issuecomment-397711962, it sounded like @konryd agreed with this and was thinking of changing this to require either plain-text or Markdown (or just Markdown since plain-text is valid Markdown).

Reply:
@prydonius I was more thinking of what @jzelinskie [noted](https://github.com/kubernetes-sigs/application/pull/48#issuecomment-398871192):

> not all tools in the pipeline are going to support rendering markdown. Importantly, if you want to front your applications with an Open Service Broker, that spec isn't going to be able to leverage your description if it's formatted in markdown.

Supporting markdown means we are adding a dependency on all tools that display this to support markdown. How does adding that dependency play out in practice?

Reply:
I was referring to this comment:
> we could end up with the case of markdown, asciidoc, and others being used. Then tools from varying places need to know how to handle this for interoperability.

As for CLI and other tooling that may not want to render Markdown, I agree that might be a good reason to keep this plain-text also. The reason we explicitly chose for NOTES.txt in Helm to be plain-text is because the main use-case was a terminal emulator where Markdown formatting is not used (that said there are tools to improve rendering in terminal emulators: https://github.com/samfoo/mdcat). That said, I think Markdown generally is readable even when not rendered, and it may be helpful for authors to use it for Notes.

Reply:
Apologies for the delay. Like @prydonius says - I'm about to change this to blessing markdown more explicitly.

I don't agree that this introduces a dependency for the tools - the idea behind markdown is to stay readable in the undrendered form. Notes rendered plainly should stay reasonably readable for humans. Out of all markup lanugages, this property of markdown is the reason to bless it. Otherwise, we'll be deciding against any sort of markup in the first place, which won't work well with UIs.